bin/publish

```diff
@@ -1,6 +1,11 @@
 #!/bin/bash -e
 
-. bin/build_utils
+CURRENT_DIR=$(cd $(dirname ${BASH_SOURCE[0]}); pwd)
```

Comment:
BASH_SOURCE is a lot more finicky and somewhat of a hack (though granted that it works in most cases):
```
BASH_SOURCE
    An array variable whose members are the source filenames where the
corresponding shell function names in the FUNCNAME array variable are
defined. The shell function ${FUNCNAME[$i]} is defined in the file 
${BASH_SOURCE[$i]} and called from ${BASH_SOURCE[$i+1]}
```
![a](https://user-images.githubusercontent.com/175811/48968339-bf272c00-efb3-11e8-9bc2-46a012d7574b.png)


I don't think I have a problem with it but you really need to quote this line properly to avoid space escapes and make sure it passes something like shellcheck.

Reply:
addressed. thanks, the quote is definitely necessary